### ISSUE TITLE
Puppeteer 1.10 stops after creating browser context on Linux

### ISSUE THREAD

**Issue by rbairwell** [puppeteer/puppeteer#3504]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 1.10.0
* Platform / OS version: Linux / Ubuntu 16.04.4 (within Docker)
* URLs (if applicable): n/a
* Node.js version: 8.12

**What steps will reproduce the problem?**

_Please include code that reproduces the issue._

```
#!/usr/bin/env node
'use strict';
const puppeteer = require('puppeteer');
console.log('loaded puppeteer');
(async () => {
    const browser = await puppeteer.launch();
    console.log('launched puppeteer');
    console.log(await browser.userAgent());
    console.log(await browser.version());
    console.log(await browser.wsEndpoint());
    console.log(await process.versions);

    browser.on('targetcreated', () => console.log('+++ Browser target created'));
    const context = await browser.createIncognitoBrowserContext();
    console.log('created context');
    const page = await context.newPage();
    console.log('created page');
    await page.goto('https://example.com');
    console.log('loaded page');
    console.log('length:'+(await page.content()).length);
    console.log('got content');
    await browser.close();
    console.log('finished');
})();

```

**What is the expected result?**
On Mac with same version of Puppeteer 1.10.0:
```
loaded puppeteer
launched puppeteer
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_1) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/72.0.3582.0 Safari/537.36
HeadlessChrome/72.0.3582.0
ws://127.0.0.1:60441/devtools/browser/5c9660e4-36e0-4e55-a14e-723b6aad3724
{ http_parser: '2.8.0',
  node: '8.11.3',
  v8: '6.2.414.54',
  uv: '1.19.1',
  zlib: '1.2.11',
  ares: '1.10.1-DEV',
  modules: '57',
  nghttp2: '1.32.0',
  napi: '3',
  openssl: '1.0.2o',
  icu: '60.1',
  unicode: '10.0',
  cldr: '32.0',
  tz: '2017c' }
created context
+++ Browser target created
created page
loaded page
length:1262
got content
finished
```

On Linux with previous version of Puppeteer 1.9.0:
```
loaded puppeteer
launched puppeteer
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/71.0.3563.0 Safari/537.36
HeadlessChrome/71.0.3563.0
ws://127.0.0.1:34785/devtools/browser/c6332edc-dbc3-4399-bf93-be582cf32d86
{ http_parser: '2.8.0',
  node: '8.12.0',
  v8: '6.2.414.66',
  uv: '1.19.2',
  zlib: '1.2.11',
  ares: '1.10.1-DEV',
  modules: '57',
  nghttp2: '1.32.0',
  napi: '3',
  openssl: '1.0.2p',
  icu: '60.1',
  unicode: '10.0',
  cldr: '32.0',
  tz: '2017c' }
created context
+++ Browser target created
created page
loaded page
length:1262
got content
finished
```

**What happens instead?**
On same Linux with Puppeteer 1.10.0:
```
loaded puppeteer
launched puppeteer
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/72.0.3582.0 Safari/537.36
HeadlessChrome/72.0.3582.0
ws://127.0.0.1:36749/devtools/browser/a4b48e44-25c5-47af-b6a9-848bf7327bfb
{ http_parser: '2.8.0',
  node: '8.12.0',
  v8: '6.2.414.66',
  uv: '1.19.2',
  zlib: '1.2.11',
  ares: '1.10.1-DEV',
  modules: '57',
  nghttp2: '1.32.0',
  napi: '3',
  openssl: '1.0.2p',
  icu: '60.1',
  unicode: '10.0',
  cldr: '32.0',
  tz: '2017c' }
created context
+++ Browser target created
```
and then hangs with Chrome still running in the background. No other output.

Launching puppeteer with dumpio provides the following:
```
launched puppeteer
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/72.0.3582.0 Safari/537.36
HeadlessChrome/72.0.3582.0
ws://127.0.0.1:38707/devtools/browser/4d0c4fd3-e30d-4d1a-ba9a-cd077b567c90
{ http_parser: '2.8.0',
  node: '8.12.0',
  v8: '6.2.414.66',
  uv: '1.19.2',
  zlib: '1.2.11',
  ares: '1.10.1-DEV',
  modules: '57',
  nghttp2: '1.32.0',
  napi: '3',
  openssl: '1.0.2p',
  icu: '60.1',
  unicode: '10.0',
  cldr: '32.0',
  tz: '2017c' }
created context
[1106/144733.934456:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. 
#0 0x55b883ccd7df <unknown>
#1 0x55b883c46f70 <unknown>
#2 0x55b88510b223 <unknown>
#3 0x55b885109424 <unknown>
#4 0x55b885107480 <unknown>
#5 0x55b887b17865 <unknown>
#6 0x55b887b1745f <unknown>
#7 0x55b8838a9b07 <unknown>
#8 0x55b8838aac8d <unknown>
#9 0x55b8838dc42b <unknown>
#10 0x55b8838a9111 <unknown>
#11 0x55b887c68488 <unknown>
#12 0x55b887c6832e <unknown>
#13 0x55b8838d96b5 <unknown>
#14 0x55b8819f11ac <unknown>
#15 0x7f60ab39c830 __libc_start_main
#16 0x55b8819f102a <unknown>

Received signal 6
#0 0x55b883ccd7df <unknown>
#1 0x55b883ccd351 <unknown>
#2 0x7f60b1514390 <unknown>
#3 0x7f60ab3b1428 gsignal
#4 0x7f60ab3b302a abort
#5 0x55b883ccc185 <unknown>
#6 0x55b883c473e2 <unknown>
#7 0x55b88510b223 <unknown>
#8 0x55b885109424 <unknown>
#9 0x55b885107480 <unknown>
#10 0x55b887b17865 content::RendererMainPlatformDelegate::EnableSandbox()
#11 0x55b887b1745f <unknown>
#12 0x55b8838a9b07 <unknown>
#13 0x55b8838aac8d msE
#14 0x55b8838dc42b <unknown>
#15 0x55b8838a9111 <unknown>
#16 0x55b887c68488 <unknown>
#17 0x55b887c6832e <unknown>
#18 0x55b8838d96b5 <unknown>
#19 0x55b8819f11ac <unknown>
#20 0x7f60ab39c830 __libc_start_main
#21 0x55b8819f102a <unknown>
  r8: 00007f60ab742770  r9: 00007f60b1a8fac0 r10: 0000000000000008 r11: 0000000000000202
 r12: 00007ffdff4261a0 r13: 0000000000000069 r14: 00007ffdff4261a8 r15: 00007ffdff4261b0
  di: 0000000000000001  si: 0000000000000001  bp: 00007ffdff425760  bx: 00007ffdff4257c0
  dx: 0000000000000006  ax: 0000000000000000  cx: 00007f60ab3b1428  sp: 00007ffdff425628
  ip: 00007f60ab3b1428 efl: 0000000000000202 cgf: 002b000000000033 erf: 0000000000000000
 trp: 0000000000000000 msk: 0000000000000000 cr2: 0000000000000000
[end of stack trace]
Calling _exit(1). Core file will not be generated.
+++ Browser target created
[1106/144733.904805:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. 
#0 0x55b883ccd7df base::debug::StackTrace::StackTrace()
#1 0x55b883c46f70 logging::LogMessage::~LogMessage()
#2 0x55b88510b223 sandbox::ProcUtil::HasOpenDirectory()
#3 0x55b885109424 service_manager::SandboxLinux::InitializeSandbox()
#4 0x55b885107480 <unknown>
#5 0x55b887b17865 <unknown>
#6 0x55b887b1745f <unknown>
#7 0x55b8838a9b07 <unknown>
#8 0x55b8838aac8d <unknown>
#9 0x55b8838dc42b <unknown>
#10 0x55b8838a9111 <unknown>
#11 0x55b887c68488 headless::(anonymous namespace)::RunContentMain()
#12 0x55b887c6832e headless::RunChildProcessIfNeeded()
#13 0x55b8838d96b5 headless::HeadlessShellMain()
#14 0x55b8819f11ac ChromeMain
#15 0x7f60ab39c830 __libc_start_main
#16 0x55b8819f102a _start

Received signal 6
#0 0x55b883ccd7df base::debug::StackTrace::StackTrace()
#1 0x55b883ccd351 base::debug::(anonymous namespace)::StackDumpSignalHandler()
#2 0x7f60b1514390 <unknown>
#3 0x7f60ab3b1428 gsignal
#4 0x7f60ab3b302a abort
#5 0x55b883ccc185 base::debug::BreakDebugger()
#6 0x55b883c473e2 logging::LogMessage::~LogMessage()
#7 0x55b88510b223 sandbox::ProcUtil::HasOpenDirectory()
#8 0x55b885109424 service_manager::SandboxLinux::InitializeSandbox()
#9 0x55b885107480 service_manager::Sandbox::Initialize()
#10 0x55b887b17865 content::RendererMainPlatformDelegate::EnableSandbox()
#11 0x55b887b1745f [1106/144734.001287:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. 
#0 0x55895e1b37df base::debug::StackTrace::StackTrace()
#1 0x55895e12cf70 logging::LogMessage::~LogMessage()
#2 0x55895f5f1223 sandbox::ProcUtil::HasOpenDirectory()
#3 0x55895f5ef424 service_manager::SandboxLinux::InitializeSandbox()
#4 0x55896168b955 content::(anonymous namespace)::ContentSandboxHelper::EnsureSandboxInitialized()
#5 0x55895f2c95ff gpu::GpuInit::InitializeAndStartSandbox()
#6 0x55896168b0d9 content::GpuMain()
#7 0x55895dd90c8d content::ContentMainRunnerImpl::Run()
#8 0x55895ddc242b service_manager::Main()
#9 0x55895dd8f111 content::ContentMain()
#10 0x55896214e488 headless::(anonymous namespace)::RunContentMain()
#11 0x55896214e32e headless::RunChildProcessIfNeeded()
#12 0x55895ddbf6b5 headless::HeadlessShellMain()
#13 0x55895bed71ac ChromeMain
#14 0x7f60d8a59830 __libc_start_main
#15 0x55895bed702a _start
```
which then seems to loop.

**Comment by jaykrz** [comment]
I spent a couple of days on this problem, dropping back to v1.9.0

**Comment by aslushnikov** [comment]
@rbairwell can you please share your docker container?

@jaykrz did you run inside docker as well?

**Comment by rbairwell** [comment]
@aslushnikov I'll see tomorrow (2330 here atm), but might not be possible as it's a company built image. The only difference in the two runs was the puppetter version - just shelled in, change package.json to fix the version, npm install and rerun script: didn't even change shell session let alone reboot etc.

**Comment by aslushnikov** [comment]
@rbairwell I see, thanks. Can you please try with puppeteer@next tomorrow? There's a newer version of chromium that might help.

**Comment by rbairwell** [comment]
Hi @aslushnikov ,

Yep, sorry, it is a custom built Docker container so wouldn't be able to share.

I did try with 'next' with practically identical results:
```
 node chrometest.js 
loaded puppeteer
[1107/102011.346494:ERROR:gpu_process_transport_factory.cc(963)] Lost UI shared context.

DevTools listening on ws://127.0.0.1:34589/devtools/browser/6d5e5ced-85e2-4f55-a52b-3284d38362d8
launched puppeteer
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/72.0.3600.0 Safari/537.36
HeadlessChrome/72.0.3600.0
ws://127.0.0.1:34589/devtools/browser/6d5e5ced-85e2-4f55-a52b-3284d38362d8
{ http_parser: '2.8.0',
  node: '8.12.0',
  v8: '6.2.414.66',
  uv: '1.19.2',
  zlib: '1.2.11',
  ares: '1.10.1-DEV',
  modules: '57',
  nghttp2: '1.32.0',
  napi: '3',
  openssl: '1.0.2p',
  icu: '60.1',
  unicode: '10.0',
  cldr: '32.0',
  tz: '2017c' }
created context
[1107/102011.486628:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. 
#0 0x56358a53b7af <unknown>
#1 0x56358a4b41a0 <unknown>
#2 0x56358b95bde3 <unknown>
#3 0x56358b959df4 <unknown>
#4 0x56358b957e00 <unknown>
#5 0x56358e3cac25 <unknown>
#6 0x56358e3ca81f <unknown>
#7 0x56358a1117c7 <unknown>
#8 0x56358a11299d <unknown>
#9 0x56358a143f9c <unknown>
#10 0x56358a110dd1 <unknown>
#11 0x56358e51a8c8 <unknown>
#12 0x56358e51a76e <unknown>
#13 0x56358a1411e5 <unknown>
#14 0x5635882461ac <unknown>
#15 0x7f7b96c74830 __libc_start_main
#16 0x56358824602a <unknown>

Received signal 6
#0 0x56358a53b7af <unknown>
#1 0x56358a53b321 <unknown>
#2 0x7f7b9cdec390 <unknown>
#3 0x7f7b96c89428 gsignal
#4 0x7f7b96c8b02a abort
#5 0x56358a53a155 <unknown>
#6 0x56358a4b462e <unknown>
#7 0x56358b95bde3 <unknown>
#8 0x56358b959df4 <unknown>
#9 0x56358b957e00 ??

#10 0x56358e3cac25 <unknown>
#11 0x56358e3ca81f <unknown>
#12 0x56358a1117c7 <unknown>
#13 0x56358a11299d <unknown>
#14 0x56358a143f9c <unknown>
#15 0x56358a110dd1 content::ContentMain()
#16 0x56358e51a8c8 <unknown>
#17 0x56358e51a76e <unknown>
#18 0x56358a1411e5 <unknown>
#19 0x5635882461ac <unknown>
#20 0x7f7b96c74830 __libc_start_main
#21 0x56358824602a <unknown>
  r8: 00007f7b9701a770  r9: 00007f7b9d367ac0 r10: 0000000000000008 r11: 0000000000000202
 r12: 00007ffdcab87be0 r13: 0000000000000069 r14: 00007ffdcab87be8 r15: 00007ffdcab87bf0
  di: 0000000000000001  si: 0000000000000001  bp: 00007ffdcab871a0  bx: 00007ffdcab871e8
  dx: 0000000000000006  ax: 0000000000000000  cx: 00007f7b96c89428  sp: 00007ffdcab87068
  ip: 00007f7b96c89428 efl: 0000000000000202 cgf: 002b000000000033 erf: 0000000000000000
 trp: 0000000000000000 msk: 0000000000000000 cr2: 0000000000000000
[end of stack trace]
Calling _exit(1). Core file will not be generated.
+++ Browser target created
[1107/102011.398970:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. 
#0 0x56358a53b7af base::debug::StackTrace::StackTrace()
#1 0x56358a4b41a0 logging::LogMessage::~LogMessage()
#2 0x56358b95bde3 sandbox::ProcUtil::HasOpenDirectory()
#3 0x56358b959df4 service_manager::SandboxLinux::InitializeSandbox()
#4 0x56358b957e00 service_manager::Sandbox::Initialize()
#5 0x56358e3cac25 pplicationCacheHostImpl21SetSubresourceFactoryEN4mojo12InterfacePtrIN7network5mojom16URLLoaderFactoryEEE
#6 0x56358e3ca81f content::RendererMain()
#7 0x56358a1117c7 content::RunZygote()
#8 0x56358a11299d content::ContentMainRunnerImpl::Run()
#9 0x56358a143f9c service_manager::Main()
#10 0x56358a110dd1 content::ContentMain()
#11 0x56358e51a8c8 headless::(anonymous namespace)::RunContentMain()
#12 0x56358e51a76e headless::RunChildProcessIfNeeded()
#13 0x56358a1411e5 headless::HeadlessShellMain()
#14 0x5635882461ac ChromeMain
#15 0x7f7b96c74830 __libc_start_main
#16 0x56358824602a _start

Received signal 6
#0 0x56358a53b7af base::debug::StackTrace::StackTrace()
#1 0x56358a53b321 base::debug::(anonymous namespace)::StackDumpSignalHandler()
#2 0x7f7b9cdec390 <unknown>
#3 0x7f7b96c89428 gsignal
#4 0x7f7b96c8b02a abort
#5 0x56358a53a155 base::debug::BreakDebugger()
#6 0x56358a4b462e logging::LogMessage::~LogMessage()
#7 0x56358b95bde3 sandbox::ProcUtil::HasOpenDirectory()
#8 0x56358b959df4 service_manager::SandboxLinux::InitializeSandbox()
#9 0x56358b957e00 service_manager::Sandbox::Initialize()
#10 0x56358e3cac25 content::RendererMainPlatformDelegate::EnableSandbox()
#11 0x56358e3ca81f content::RendererMain()
#12 0x56358a1117c7 content::RunZygote()
#13 0x56358a11299d content::ContentMainRunnerImpl::Run()
#14 0x56358a143f9c service_manager::Main()
#15 0x56358a110dd1 content::ContentMain()
#16 0x56358e51a8c8 [1107/102011.608758:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. 
#0 0x55ac48dc77af base::debug::StackTrace::StackTrace()
#1 0x55ac48d401a0 logging::LogMessage::~LogMessage()
#2 0x55ac4a1e7de3 sandbox::ProcUtil::HasOpenDirectory()
#3 0x55ac4a1e5df4 service_manager::SandboxLinux::InitializeSandbox()
#4 0x55ac4c2d30e5 content::(anonymous namespace)::ContentSandboxHelper::EnsureSandboxInitialized()
#5 0x55ac49ef51ec gpu::GpuInit::InitializeAndStartSandbox()
#6 0x55ac4c2d2869 content::GpuMain()
#7 0x55ac4899e99d content::ContentMainRunnerImpl::Run()
#8 0x55ac489cff9c service_manager::Main()
#9 0x55ac4899cdd1 content::ContentMain()
#10 0x55ac4cda68c8 headless::(anonymous namespace)::RunContentMain()
#11 0x55ac4cda676e headless::RunChildProcessIfNeeded()
#12 0x55ac489cd1e5 headless::HeadlessShellMain()
#13 0x55ac46ad21ac ChromeMain
#14 0x7f11e8728830 __libc_start_main
#15 0x55ac46ad202a _start

Received signal 6
#0 0x55ac48dc77af headless::(anonymous namespace)::RunContentMain()
#17 0x56358e51a76e base::debug::StackTrace::StackTrace()
#1 0x55ac48dc7321 base::debug::(anonymous namespace)::StackDumpSignalHandler()
#2 0x7f11ee8a0390 <unknown>
#3 0x7f11e873d428 gsignal
#4 0x7f11e873f02a abort
#5 0x55ac48dc6155 headless::RunChildProcessIfNeeded()
#18 0x56358a1411e5 headless::HeadlessShellMain()
#19 0x5635882461ac base::debug::BreakDebugger()
#6 0x55ac48d4062e logging::LogMessage::~LogMessage()
#7 0x55ac4a1e7de3 ChromeMain
#20 0x7f7b96c74830 __libc_start_main
#21 0x56358824602a sandbox::ProcUtil::HasOpenDirectory()
#8 0x55ac4a1e5df4 service_manager::SandboxLinux::InitializeSandbox()
#9 0x55ac4c2d30e5 _start
  r8: 00007f7b9701a770  r9: 00007f7b9d367ac0 r10: 0000000000000008 r11: 0000000000000202
 r12: 00007ffdcab87be0 r13: 0000000000000069 r14: 00007ffdcab87be8 r15: 00007ffdcab87bf0
  di: 0000000000000001  si: 0000000000000001  bp: 00007ffdcab871a0  bx: 00007ffdcab871e8
  dx: 0000000000000006  ax: 0000000000000000  cx: 00007f7b96c89428  sp: 00007ffdcab87068
  ip: 00007f7b96c89428 efl: 0000000000000202 cgf: 002b000000000033 erf: 0000000000000000
 trp: 0000000000000000 msk: 0000000000000000 cr2: 0000000000000000
[end of stack trace]
Calling _exit(1). Core file will not be generated.
content::(anonymous namespace)::ContentSandboxHelper::EnsureSandboxInitialized()
#10 0x55ac49ef51ec gpu::GpuInit::InitializeAndStartSandbox()
#11 0x55ac4c2d2869 content::GpuMain()
#12 0x55ac4899e99d content::ContentMainRunnerImpl::Run()
#13 0x55ac489cff9c service_manager::Main()
#14 0x55ac4899cdd1 content::ContentMain()
#15 0x55ac4cda68c8 headless::(anonymous namespace)::RunContentMain()
#16 0x55ac4cda676e headless::RunChildProcessIfNeeded()
#17 0x55ac489cd1e5 headless::HeadlessShellMain()
#18 0x55ac46ad21ac ChromeMain
#19 0x7f11e8728830 __libc_start_main
#20 0x55ac46ad202a _start
  r8: 00007f11e8ace770  r9: 00007f11eee1bac0 r10: 0000000000000008 r11: 0000000000000202
 r12: 00007fff58f608e0 r13: 0000000000000069 r14: 00007fff58f608e8 r15: 00007fff58f608f0
  di: 00000000000000b3  si: 00000000000000b3  bp: 00007fff58f5fea0  bx: 00007fff58f5fee8
  dx: 0000000000000006  ax: 0000000000000000  cx: 00007f11e873d428  sp: 00007fff58f5fd68
  ip: 00007f11e873d428 efl: 0000000000000202 cgf: 002b000000000033 erf: 0000000000000000
 trp: 0000000000000000 msk: 0000000000000000 cr2: 0000000000000000
[end of stack trace]
Calling _exit(1). Core file will not be generated.
[1107/102012.237164:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. 
#0 0x558ae7d317af base::debug::StackTrace::StackTrace()
#1 0x558ae7caa1a0 logging::LogMessage::~LogMessage()
#2 0x558ae9151de3 sandbox::ProcUtil::HasOpenDirectory()
#3 0x558ae914fdf4 service_manager::SandboxLinux::InitializeSandbox()
#4 0x558aeb23d0e5 content::(anonymous namespace)::ContentSandboxHelper::EnsureSandboxInitialized()
#5 0x558ae8e5f1ec gpu::GpuInit::InitializeAndStartSandbox()
#6 0x558aeb23c869 content::GpuMain()
#7 0x558ae790899d content::ContentMainRunnerImpl::Run()
#8 0x558ae7939f9c service_manager::Main()
#9 0x558ae7906dd1 content::ContentMain()
#10 0x558aebd108c8 headless::(anonymous namespace)::RunContentMain()
#11 0x558aebd1076e headless::RunChildProcessIfNeeded()
#12 0x558ae79371e5 headless::HeadlessShellMain()
#13 0x558ae5a3c1ac ChromeMain
#14 0x7f63884b7830 __libc_start_main
#15 0x558ae5a3c02a _start

Received signal 6
#0 0x558ae7d317af base::debug::StackTrace::StackTrace()
#1 0x558ae7d31321 base::debug::(anonymous namespace)::StackDumpSignalHandler()
#2 0x7f638e62f390 <unknown>
```
etc.

I note that https://try-puppeteer.appspot.com/ is still running 1.9.0, but I've just cloned the try-puppeteer repo from https://github.com/ebidel/try-puppeteer (which comes with it's own Docker setup) and get the same results as 1.10.0:
```
$ ./backend/run_puppeteer.sh chrometest.js 
Running chrometest.js in Puppeteer...
loaded puppeteer
Fontconfig warning: "/etc/fonts/fonts.conf", line 86: unknown element "blank"
[1107/103233.755033:ERROR:gpu_process_transport_factory.cc(967)] Lost UI shared context.

DevTools listening on ws://127.0.0.1:38805/devtools/browser/8a32d1ef-35e0-4e49-a7b0-2b030d024d89
launched puppeteer
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/72.0.3582.0 Safari/537.36
HeadlessChrome/72.0.3582.0
ws://127.0.0.1:38805/devtools/browser/8a32d1ef-35e0-4e49-a7b0-2b030d024d89
{ http_parser: '2.8.0',
  node: '8.12.0',
  v8: '6.2.414.66',
  uv: '1.19.2',
  zlib: '1.2.11',
  ares: '1.10.1-DEV',
  modules: '57',
  nghttp2: '1.32.0',
  napi: '3',
  openssl: '1.0.2p',
  icu: '60.1',
  unicode: '10.0',
  cldr: '32.0',
  tz: '2017c' }
created context
[1107/103233.883411:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. 
#0 0x5603466da7df base::debug::StackTrace::StackTrace()
#1 0x560346653f70 logging::LogMessage::~LogMessage()
#2 0x560347b18223 x7bpf_dsl4TrapEPFlRKNS_17arch_seccomp_dataEPvEPKv
#3 0x560347b16424 <unknown>
#4 0x560347b14480 <unknown>
#5 0x56034a524865 <unknown>
#6 0x56034a52445f <unknown>
#7 0x5603462b6b07 <unknown>
#8 0x5603462b7c8d <unknown>
#9 0x5603462e942b <unknown>
#10 0x5603462b6111 <unknown>
#11 0x56034a675488 <unknown>
#12 0x56034a67532e <unknown>
#13 0x5603462e66b5 <unknown>
#14 0x5603443fe1ac <unknown>
#15 0x7f372db05b45 __libc_start_main
#16 0x5603443fe02a <unknown>

Received signal 6
#0 0x5603466da7df [1107/103233.850463:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. 
#0 0x5603466da7df base::debug::StackTrace::StackTrace()
#1 0x560346653f70 logging::LogMessage::~LogMessage()
#2 0x560347b18223 <unknown>
#3 0x560347b16424 <unknown>
#4 0x560347b14480 <unknown>
#5 0x56034a524865 <unknown>
#6 0x56034a52445f <unknown>
#7 0x5603462b6b07 <unknown>
#8 0x5603462b7c8d <unknown>
#9 0x5603462e942b <unknown>
#10 0x5603462b6111 <unknown>
#11 0x56034a675488 <unknown>
#12 0x56034a67532e <unknown>
#13 0x5603462e66b5 <unknown>
#14 0x5603443fe1ac <unknown>
#15 0x7f372db05b45 __libc_start_main
#16 0x5603443fe02a <unknown>

Received signal 6
#0 0x5603466da7df <unknown>
#1 0x5603466da351 <unknown>
#2 0x7f3733bb7890 <unknown>
#3 0x7f372db19067 gsignal
#4 0x7f372db1a448 abort
#5 0x5603466d9185 <unknown>
#6 0x5603466543e2 <unknown>
#7 0x560347b18223 <unknown>
#8 0x560347b16424 <unknown>
#9 0x560347b14480 <unknown>
#10 0x56034a524865 U

#1 0x5603466da351 base::debug::(anonymous namespace)::StackDumpSignalHandler()
#2 0x7f3733bb7890 <unknown>
#3 0x7f372db19067 gsignal
#4 0x7f372db1a448 abort
#5 0x5603466d9185 <unknown>
#6 0x5603466543e2 <unknown>
#7 0x560347b18223 <unknown>
#8 0x560347b16424 <unknown>
#9 0x560347b14480 <unknown>
#10 0x56034a524865 <unknown>
#11 0x56034a52445f <unknown>
#12 0x5603462b6b07 <unknown>
#13 0x5603462b7c8d <unknown>
#14 0x5603462e942b <unknown>
#15 0x5603462b6111 <unknown>
#16 0x56034a675488 <unknown>
#17 0x56034a67532e <unknown>
#18 0x5603462e66b5 <unknown>
#19 0x5603443fe1ac <unknown>
#20 0x7f372db05b45 __libc_start_main
#21 0x5603443fe02a <unknown>
  r8: 00007f37341bda00  r9: 00003db242ab8b80 r10: 0000000000000008 r11: 0000000000000202
 r12: 00007ffcdc34c5f0 r13: 0000000000000069 r14: 00007ffcdc34c5f8 r15: 00007ffcdc34c600
  di: 0000000000000001  si: 0000000000000001  bp: 00007ffcdc34bbb0  bx: 00007ffcdc34bc10
  dx: 0000000000000006  ax: 0000000000000000  cx: 00007f372db19067  sp: 00007ffcdc34ba78
  ip: 00007f372db19067 efl: 0000000000000202 cgf: 002b000000000033 erf: 0000000000000000
 trp: 0000000000000000 msk: 0000000000000000 cr2: 0000000000000000
[end of stack trace]
Calling _exit(1). Core file will not be generated.
+++ Browser target created
content::RendererMainPlatformDelegate::EnableSandbox()
#11 0x56034a52445f content::RendererMain()
#12 0x5603462b6b07 content::RunZygote()
#13 0x5603462b7c8d content::ContentMainRunnerImpl::Run()
#14 0x5603462e942b service_manager::Main()
#15 0x5603462b6111 content::ContentMain()
#16 0x56034a675488 headless::(anonymous namespace)::RunContentMain()
#17 0x56034a67532e headless::RunChildProcessIfNeeded()
#18 0x5603462e66b5 headless::HeadlessShellMain()
#19 0x5603443fe1ac ChromeMain
#20 0x7f372db05b45 __libc_start_main
#21 0x5603443fe02a _start
  r8: 00007f37341bda00  r9: 00003db242ab8b80 r10: 0000000000000008 r11: 0000000000000202
 r12: 00007ffcdc34c5f0 r13: 0000000000000069 r14: 00007ffcdc34c5f8 r15: 00007ffcdc34c600
  di: 0000000000000001  si: 0000000000000001  bp: 00007ffcdc34bbb0  bx: 00007ffcdc34bc10
  dx: 0000000000000006  ax: 0000000000000000  cx: 00007f372db19067  sp: 00007ffcdc34ba78
  ip: 00007f372db19067 efl: 0000000000000202 cgf: 002b000000000033 erf: 0000000000000000
 trp: 0000000000000000 msk: 0000000000000000 cr2: 0000000000000000
[end of stack trace]
Calling _exit(1). Core file will not be generated.
[1107/103233.911269:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. 
#0 0x559d7be8d7df base::debug::StackTrace::StackTrace()
```

**Comment by jaykrz** [comment]
> @jaykrz did you run inside docker as well?

@aslushnikov, yes running inside a docker container

Slightly different for me, we are generating pdfs and I noticed that when going to a URL that has some js trying to manipulate the drawing context of a canvas element it would hang with the similar results in the log.

Dropping back to 1.9.0 stops the problem from happening. Sorry, I also can't share our Dockerfile. I can try to see if I can repro on a simplified version of it.

**Comment by jaykrz** [comment]
@aslushnikov
I added a Dockerfile and puppeteer file to a new issue, I don't want to take over this one since my problem is a bit different.

https://github.com/GoogleChrome/puppeteer/issues/3516

**Comment by rbairwell** [comment]
@jaykrz -

Just at a glance, they do look to be the same (or closely related) issues - both hitting a fatal at: `[1108/144156.157131:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. `
I know very very little about the Chromium/Puppeteer source code (and I've yet to test this sans-Puppeteer with just devtools on Chromium 72.0.3582.0 on Linux under Docker)

**Comment by aslushnikov** [comment]
@rbairwell @jaykrz are you guys running single process? In which environment do you run this dockers - is it AWS?

**Comment by hyperknot** [comment]
I have run into this same regression. Host is dedicated server (no virtualization) Ubuntu 16.04. Docker guest is Ubuntu 16.04.

Deploying 1.10 put system CPU load to 200%. Interestingly not user, but system. Normally puppeteer raises the user CPU.

Reverting to 1.9 fixed this.

Graph of CPU usage while having 1.10 deployed:
![image](https://user-images.githubusercontent.com/494223/48303743-906a7980-e50e-11e8-9b1f-293ad99cfe45.png)

The blue bit at the end is the fixed 1.9 working on all the missed background tasks, and is how it should look.

**Comment by aslushnikov** [comment]
@hyperknot can you please try `pupppeteer@latest` and see if the problem persists? Looks like some regression in headless.

**Comment by hyperknot** [comment]
@aslushnikov isn't latest the same as 1.10.0?
https://www.npmjs.com/package/puppeteer

**Comment by aslushnikov** [comment]
@hyperknot oops I meant `puppeteer@next` - that'll be Puppeteer tip-of-tree

**Comment by hyperknot** [comment]
@aslushnikov I tried `@next` as of today, and the results were the same. 200% system CPU, not working.

**Comment by aslushnikov** [comment]
@hyperknot that's quite weird. Works fine on my local ubuntu, as well as on Travis (virtual machine) and Cirrus (docker + gcloud). Anything specific about your setup?

**Comment by hyperknot** [comment]
The only not common part of my code is that I'm using these parameters to allow capturing a MapboxGL (WebGL) map:

```
puppeteer.launch({
    headless: false,
    args: ['--headless', '--disable-dev-shm-usage', '--no-sandbox'],
  })
```

**Comment by aslushnikov** [comment]
@hyperknot any chance you can share your whole script so that I can run it locally?

**Comment by hyperknot** [comment]
@aslushnikov 

Here is the script, which reproduces this on Docker / Linux:
```
const puppeteer = require('puppeteer')

//  works
// const url = 'https://wikipedia.org'

// frozen, 100% CPU
const url =
  'https://api.mapbox.com/styles/v1/mapbox/streets-v9.html?title=true&access_token=pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpejY4M29iazA2Z2gycXA4N2pmbDZmangifQ.-g_vE53SD2WrJ6tFX7QHmA#1.07/-0.0/0.0'

async function run() {
  const browser = await puppeteer.launch({
    headless: false,
    args: ['--headless', '--disable-dev-shm-usage', '--no-sandbox'],
  })

  const page = await browser.newPage()
  await page.setViewport({
    width: 600,
    height: 400,
  })

  try {
    await page.goto(url, { waitUntil: 'networkidle0', timeout: 20 * 1000 })
  } catch (error) {
    console.error(error)
  }

  await page.screenshot({
    path: 'out.png',
  })

  await browser.close()
}

run()
```

The key is something to do with GPU or WebGL. The wikipedia url works, the Mapbox GL one does not. macOS is not affected by this bug, it works well for me.

There is a timeout error, but it's only pointing to the page.goto code.

```
{ TimeoutError: Navigation Timeout Exceeded: 20000ms exceeded
    at Promise.then (/home/app/web/build/node_modules/puppeteer/lib/FrameManager.js:1230:21)
    at <anonymous>
  -- ASYNC --
    at Frame.<anonymous> (/home/app/web/build/node_modules/puppeteer/lib/helper.js:144:27)
    at Page.goto (/home/app/web/build/node_modules/puppeteer/lib/Page.js:587:49)
    at Page.<anonymous> (/home/app/web/build/node_modules/puppeteer/lib/helper.js:145:23)
    at run (/shared/render_test.js:23:16)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7) name: 'TimeoutError' }
```

**Comment by jukra** [comment]
I was also encountering weird errors on a page where we are using Chart.js to render charts. I was able to render the page if I ran to code locally, but when using Docker, got Timeout issues without clear indication on the error itself. If I tried to render google.com for example, it worked fine on both cases. After launching Puppeteer with `dumpio: true` I saw the same error @rbairwell is seeing.

Thanks to tips on this thread, I downgraded to Puppeteer 1.9.0 and everything is working fine. I also tried using `puppeteer@next` but had the same issue with that one as well.

My Dockerfile is almost the same as [browserless Dockerfile](https://github.com/joelgriffith/browserless/blob/master/Dockerfile), but with a few tweaks related to our application.

**Comment by donovan-graham** [comment]
Just upgraded and seeing the same error when running headless chrome in an alpine dockerized image.  This happens before running puppeteer, so I'm assuming its something to do with chrome or one of the run flags I'm using. 

puppeteer@v1.10.0
Chrome r599821
Chrome 72.0.3582.0

And 

puppeteer@next
Chrome r609904


```
chrome_1  | DevTools listening on ws://0.0.0.0:9222/devtools/browser/7f971e7f-d6b4-470d-900a-5a55f563e584
chrome_1  | [1127/201710.967722:FATAL:proc_util.cc(97)] Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0. 
chrome_1  | #0 0x55d26316387f base::debug::StackTrace::StackTrace()
chrome_1  | #1 0x55d2630daa60 logging::LogMessage::~LogMessage()
chrome_1  | #2 0x55d26458d4a3 sandbox::ProcUtil::HasOpenDirectory()
chrome_1  | #3 0x55d26458b4b4 service_manager::SandboxLinux::InitializeSandbox()
chrome_1  | #4 0x55d2666bce39 content::(anonymous namespace)::ContentSandboxHelper::EnsureSandboxInitialized()
chrome_1  | #5 0x55d2644c8d73 gpu::GpuInit::InitializeAndStartSandbox()
chrome_1  | #6 0x55d2666bc621 content::GpuMain()
chrome_1  | #7 0x55d262d3a40d content::ContentMainRunnerImpl::Run()
chrome_1  | #8 0x55d262d6b9f6 service_manager::Main()
chrome_1  | #9 0x55d262d388a1 content::ContentMain()
chrome_1  | #10 0x55d26716ebb8 headless::(anonymous namespace)::RunContentMain()
chrome_1  | #11 0x55d26716ea5e headless::RunChildProcessIfNeeded()
chrome_1  | #12 0x55d262d68c05 headless::HeadlessShellMain()
chrome_1  | #13 0x55d260e351ac ChromeMain
chrome_1  | #14 0x7f2c9d99c2e1 __libc_start_main
chrome_1  | #15 0x55d260e3502a _start

.....

chrome_1  | Calling _exit(1). Core file will not be generated.

```

**Comment by donovan-graham** [comment]
Update: this error seems to only appear when I'm running with the `--disable-dev-shm-usage` flag.

**Comment by brockfanning** [comment]
Just a note that I'm having a similar experience as @jukra:
*  also trying to test a page that uses Chart.js.
* works fine locally
* when run on [CircleCI's prebuilt Docker image](https://circleci.com/docs/2.0/circleci-images/) `circleci/ruby:2.5.1-node-browsers`:
    * fails with the "Error: function timed out" errors
    * with `dumpio` at `true`, shows the same errors as @rbairwell 
* downgrading to Puppeteer 1.9.0 resolves it

**Comment by samylaumonier** [comment]
Same issue, everything is working fine when I run `circleci local execute --job build` locally (Fedora 28, docker image `circleci/node:latest-browsers`), but on CircleCI I have the same fatal error:

>Check failed: fstatat(proc_self_fd, de->d_name, &s, 0) == 0.

For my part, downgrading to puppeteer@1.9.0 (Chromium r594312) does not solve the issue. And I am not using the `--disable-dev-shm-usage` flag.

EDIT: Not working with puppeteer@next (Chromium r615489) neither.

**Comment by Jivings** [comment]
Same issue with CircleCI, downgrading to 1.9.0 fixes.

**Comment by aadityataparia** [comment]
This Fixes it for me: You need to match puppeteer chromium version with chrome version on circleci.

check chrome version on circleci with `google-chrome --version`

**Comment by fbkaminski** [comment]
If anyone is still wondering about this: Im my case im not using puppeter, but through googling the error this page was a hit. (In my case im developing something based on the chromium source)

Anyway, im getting the same error while trying to run my chromium based executable through Windows WSL.

The code at fault is here: 
https://github.com/chromium/chromium/blob/master/sandbox/linux/services/proc_util.cc#L97

Its the gpu sandbox that is performing this check, to see if theres open dirs on the sandboxed process before starting the sandbox.

I´ve tried a simple code, almost the same as the code from chrome to test if its a generic thing, and it turns out that launching the executable by hand return 0  on 'fstatat()' as expected, so i added a code to print the error linux (inside WSL) is getting to see whats happening.. My guess is that is something to do with some permissions the GPU process does not have.

Will follow with the real answer to anyone also getting this error. I hope it helps.

**Comment by fbkaminski** [comment]
Just to leave the answer here: yes, its lack of enough permissions to use fstatat() on the dirs of 'self/fd/' by the gpu process.

Given my test, launching the same code but in a binary launched by hand had enough permissions, i guess that the problem would be a third-party process launching the gpu process that would not have enough permissions in WSL (or Docker) , like the SUID process launcher of chrome.

For now, given i countrol the source code, i have just disabled this check, and i can keep the GPU process running.. but to solve this problem in the long term i dont know what Docker or WSL do differently from vanilla Linux that this sandbox check are not able to continue on its own (in Linux this was never a issue).

Anyway, whatever process launching the gpu process, needs to grant this permission to its child, so the gpu can make this check without any issue, or maybe this is something badly wired on WSL implementation for instance, that should be addressed with a patch on WSL itself.

**Comment by AlonMiz** [comment]
+1
inside docker
```
Failed to launch the browser process!
Fontconfig warning: "/etc/fonts/fonts.conf", line 100: unknown element "blank"
[0316/124325.728121:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
#26 38.60     Inconsistency detected by ld.so: ../elf/dl-tls.c: 493: _dl_allocate_tls_init: Assertion `listp->slotinfo[cnt].gen <= GL(dl_tls_generation)' failed!
TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md
at onClose (node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:193:20)
at ChildProcess.<anonymous> (node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:184:79)
```
